rest_framework/fields.py

```diff
@@ -1398,6 +1399,9 @@ def to_internal_value(self, data):
         if data == '' and self.allow_blank:
             return ''
 
+        if isinstance(data, IntegerChoices) and str(data) != str(data.value):
```

Comment:
i'm happy with the handling of IntegerChoices but I would also love to see TextChoices and other buiilt in django enum choices compats as well

Reply:
@auvipy I've added `TextChoices` here too but should we have `Enum` too?